enzy_htp/preparation/align_ligand.py

```diff
@@ -0,0 +1,196 @@
+"""Functionality for aligning the coordinates of a ligand to a target structure using mapped PDB-style atom names. Note that atoms 
+without names will not be easily aligned. Only align_ligand() should be called. All other functions are for implementation purposes 
+only and SHOULD NOT be called directly by the user.
+
+Author: Chris Jurich <chris.jurich@vanderbilt.edu>
+Date: 2023-10-09
+"""
+
+from collections import defaultdict
+from typing import List, Dict, Set, Tuple
+from pathlib import Path
+
+import pandas as pd
+import numpy as np
+import rdkit.Chem as _rchem
+from rdkit.Chem import rdMolTransforms as rdmt
+
+from enzy_htp import config, interface
+
+from enzy_htp.core import file_system as fs
+from enzy_htp.core import _LOGGER
+
+
+def align_ligand(template: str,
```

Comment:
We should have most thing outside of the _interface take and output objects in /strcuture.
In this way the protocol is not coupled with a specific file type. 
For this function, it will be great if they IOs are Ligand()s

Reply:
Looking more into this make me feel the current version of the function should go to RDKit interface and leave just an engining use "science API" here

Reply:
Due to how this alignment algorithm functions, the input file format HAS to be `.mol2`. This is because the function requires connectivity, bond type, and PDB-style atom naming. There is no other format that allows for this. I am not sure where this should go. The issue is that this function is not a true "alignment"; it uses PDB atom naming to align heavy atoms. I probably just need to put a more descriptive name.

Reply:
I think `.pdb` also fits these requirements lol.
For what you describe, you can put this in `structure_operation/ligand.py` or `preparation/reactive_docking` (is it only going to ever be used by EnzyRCD)